meta-mel/recipes-core/meta/meta-environment.bbappend

```diff
@@ -5,3 +5,23 @@ python set_multilib_variants () {
         d.setVar('MULTILIB_VARIANTS', variants)
 }
 do_generate_content[prefuncs] += "set_multilib_variants"
+
+create_sdk_files_append_feature-mentor-staging () {
+    script=${SDK_OUTPUT}/${SDKPATH}/environment-setup-${REAL_MULTIMACH_TARGET_SYS}
+    cat >"${script}.new" <<END
+if [ -n "\$BASH_SOURCE" ] || [ -n "\$ZSH_NAME" ]; then
+    if [ -n "\$BASH_SOURCE" ]; then
+        scriptdir="\$(cd "\$(dirname "\$BASH_SOURCE")" && pwd)"
+    elif [ -n "\$ZSH_NAME" ]; then
+        scriptdir="\$(cd "\$(dirname "\$0")" && pwd)"
+    fi
+else
+    if [ ! -d "${SDKPATH}" ]; then
+        echo >&2 "Warning: Unable to determine SDK install path from environment setup script location, using default of ${SDKPATH}."
+    fi
+    scriptdir="${SDKPATH}"
+fi
+END
+    sed -e "s#${SDKPATH}#\$scriptdir#g" "$script" >>"${script}.new"
+    mv "${script}.new" "${script}"
+}
```

Comment:
If CB is the only case we need to support this with, wouldn't meta-mentor-private be a better host for this?

Reply:
Hmm that's a good point, thanks. I tend to think of that layer as 'ade', while this technically isn't ade-specific, it does serve the same goal, of supporting CB. I'll move it there once I verify that it's branched.

Reply:
I'm going to take two steps here:
1. Move this bbappend to meta-mentor-private
2. Also modify oe-core's meta-environment to include this, on a branch on oe-core-contrib for a future effort to improve the relocation handling of the sdk, so it doesn't completely fall off my radar

Reply:
Submitted https://github.com/MentorEmbedded/mel-meta-mentor-private/pull/146 for master there and removed the append entirely here.